src/aspire/operators/blk_diag_matrix.py

```diff
@@ -398,10 +398,12 @@ def matmul(self, other, inplace=False):
         """
 
         if not isinstance(other, BlkDiagMatrix):
-            raise RuntimeError(
-                "Attempt BlkDiagMatrix matrix multiplication "
-                "(matmul,@) of non BlkDiagMatrix {}, try (*,mul)".format(repr(other))
-            )
+            if inplace:
+                raise RuntimeError(
+                    "`inplace` method not supported when "
+                    "mixing `BlkDiagMatrix` and `Numpy`."
+                )
+            return self.apply(other)
```

Comment:
Just so I understand, this is the added syntactic sugar discussed in the description – allowing us to call matmul or @ instead of invoking `apply`?

Reply:
Yep.